spark-statsd-reporter/src/main/java/org/apache/spark/metrics/sink/statsd/StatsdReporter.java

```diff
@@ -0,0 +1,189 @@
+            this.registry = registry;
+        }
+
+        Builder formatter(MetricFormatter metricFormatter) {
+            this.metricFormatter = metricFormatter;
+            return this;
+        }
+
+        Builder convertRatesTo(TimeUnit rateUnit) {
+            this.rateUnit = rateUnit;
+            return this;
+        }
+
+        Builder convertDurationsTo(TimeUnit durationUnit) {
+            this.durationUnit = durationUnit;
+            return this;
+        }
+
+        public Builder filter(MetricFilter filter) {
+            this.filter = filter;
+            return this;
+        }
+
+        Builder host(String host) {
+            this.host = host;
+            return this;
+        }
+
+        Builder port(int port) {
+            this.port = port;
+            return this;
+        }
+
+        StatsdReporter build() {
+            return new StatsdReporter(registry, metricFormatter, reporterName, rateUnit, durationUnit, filter, host, port);
+        }
+    }
+
+    @Override
+    public void report(SortedMap<String, Gauge> gauges, SortedMap<String, Counter> counters, SortedMap<String, Histogram> histograms, SortedMap<String, Meter> meters, SortedMap<String, Timer> timers) {
+        gauges.keySet().forEach(System.out::println);
+        counters.keySet().forEach(System.out::println);
+        histograms.keySet().forEach(System.out::println);
+        meters.keySet().forEach(System.out::println);
+        timers.keySet().forEach(System.out::println);
+
+        try (DatagramSocket socket = new DatagramSocket()) {
+            try {
+                reportGauges(gauges, socket);
+                reportCounters(counters, socket);
+                reportHistograms(histograms, socket);
+                reportMeters(meters, socket);
+                reportTimers(timers, socket);
+            } catch (StatsdReporterException e) {
+                logger.warn("Unable to send packets to StatsD", e);
+            }
+        } catch (IOException e) {
+            logger.warn("StatsD datagram socket construction failed", e);
+        }
+    }
+
+    @VisibleForTesting
+    void reportGauges(SortedMap<String, Gauge> gauges, DatagramSocket socket) {
+        gauges.forEach((name, value) ->
+                send(socket, metricFormatter.buildMetricString(name, value.getValue(), GAUGE))
+        );
+    }
+
+    @VisibleForTesting
+    void reportCounters(SortedMap<String, Counter> counters, DatagramSocket socket) {
+        counters.forEach((name, value) ->
+                send(socket, metricFormatter.buildMetricString(name, value.getCount(), COUNTER))
+        );
+    }
+
+    @VisibleForTesting
+    void reportHistograms(SortedMap<String, Histogram> histograms, DatagramSocket socket) {
+        histograms.forEach((name, histogram) -> {
+            Snapshot snapshot = histogram.getSnapshot();
+            send(socket,
+                    metricFormatter.buildMetricString(name, "count", histogram.getCount(), GAUGE),
```

Comment:
Sounds great, but we depend on Spark transitive Dropwizard dependency which doesn't have it in `3.1.5`. Depending on Spark this way allows us to avoid any version conflicts in libraries.

Reply:
In this case, we won't be able to use latest version of `dropwizard-metrics` for filtering. It uses `MetricAttribute` in built for defining set of attributes.

Reply:
Good point. It looks like `dropwizard-metrics` requires at least `metrics-core:4.02`

Reply:
I suggest doing the following: separately bump the version of `metrics-core` in mesosphere/spark to the latest `4.0.5` as per https://github.com/dropwizard/metrics/releases and then add necessary dependencies to statsd-reporter to minimize the impact on Spark codebase itself.